### PR TITLE
Add GitHub Actions workflow

### PR DESCRIPTION
**Pull request by alexashley**
Workflow:
- evaluate policy on any push
- if we're on dev, staging, or prod branches do a deployment
- don't require dev to pass policy

There's also some updates in here to the chart -- reference the image pull secret created by https://github.com/rode/demo/pull/47 and deploy the actual app image instead of nginx

### CODE REVIEW

File: .github/workflows/policy.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        uses: rode/evaluate-policy-action@v0.1.0
+        with:
+          enforce: ${{ github.ref != 'refs/heads/dev' }} # don't require policy to pass in dev
+          policyName: Sample Harbor Policy
+          resourceUri: ${{ steps.get.outputs.resourceUri }}
+          rodeHost: ${{ env.RODE_HOST }}
+          rodeInsecure: true
+
+  deploy:
+    if: ${{ contains(fromJson('["refs/heads/dev", "refs/heads/staging", "refs/heads/prod"]'), github.ref) }}
```
Comment (alexashley) [inline .github/workflows/policy.yaml:37]: As an alternative to this, we could split the deployment into its own workflow, but I couldn't think of a good way to do that without duplicating the policy block

File: charts/demo-app/templates/deployment.yaml [hunk 2]
```diff
@@ -31,7 +31,7 @@ spec:
         - name: {{ .Chart.Name }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
-          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          image: "{{ .Values.image.repository }}@sha256:{{ .Values.image.tag }}"
```
Comment (alexashley) [inline charts/demo-app/templates/deployment.yaml:34]: Technically what we've been passing as `image.tag` from the `demo-app` has never been a tag, but instead the SHA 256 digest. 

I made the change here to deploy that instead, although the fact that it's called a tag is still misleading. We can update the `demo-app` pipeline to pass both the tag and the digest if we want to fix that.

File: .github/workflows/policy.yaml (hunk 1 above)
Comment (RobertKelly) [reply .github/workflows/policy.yaml:37]: I thought about that - may make sense later.

File: .github/workflows/policy.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    needs:
+      - evaluate
```
Comment (RobertKelly) [inline .github/workflows/policy.yaml:39]: This was an important add.  Dependency on eval.

**Review by RobertKelly — APPROVED** [review]

**Pull request merged**